tree/tree/src/TChain.cxx

```diff
@@ -2696,7 +2696,10 @@ void TChain::SetEntryList(TEntryList *elist, Option_t *opt)
       auto chainElement = (TChainElement*)fFiles->UncheckedAt(ie);
       treename = chainElement->GetName();
       filename = chainElement->GetTitle();
-      templist = elist->GetEntryList(treename, filename, opt);
+      // templist = elist->GetEntryList(treename, filename, opt);
+      // TODO: Add checks that the sub TEntryList is referring to the
+      // correct, current, tree in the TChain.
+      templist = static_cast<TEntryList *>(elist->GetLists()->At(ie));
```

Comment:
This assumes that there is exactly one sub-entrylist per files, is that always the case (ie. can there be 2 partial list for a single file or could there be files skipped?)

Reply:
The current behaviour is already making assumptions, though now that you mention
> could there be files skipped

I realize that we'd be trading assumptions, let me elaborate.
Currently we don't support the case where the TChain is made of the same tree and file repeated multiple times. But at the same time, if there were a TChain made of 3 different trees, and we wanted to have TEntrylists only for two of them, this would work because `GetEntryList(treename, filename)` would recognize the two trees of which we want to select entries (provided each of them has a unique treename, filename pair)

With the current status of this commit, we would be supporting the first case above while losing the latter because as you wrote it assumes there is exactly one sub list per file in the chain.

As for the other case you mention
> can there be 2 partial list for a single file

I believe this is not supported also on current master, as the two partial list for a single file would behave like the linked issue (two lists with same treename, filename)

Reply:
It seems to me that the currently supported case (skipping a file, likely because none of the entries are selected) would be more common that the new case (using the same file multiple times to increase statistics by using duplication), what do you think?

Reply:
I agree. And in general I wouldn't push a "fix" if it stopped supporting a preexisting use case. I will try to find a way to support both cases :+1: